### ISSUE TITLE
Mouse Coordinate Offset Issue in Fullscreen Mode on MacBook with Notch

### ISSUE THREAD

**Issue by qqiu-cyt** [libsdl-org/SDL#10441]
I am encountering an issue on a MacBook model that includes a screen notch in fullscreen mode. After switching an SDL window to fullscreen mode, there is a noticeable offset between the actual position of the mouse pointer and the coordinates where the SDL window registers mouse events.

**Comment by slouken** [comment]
Are you seeing this on SDL3? How are you setting fullscreen mode? Are you able to reproduce this with the testwm test program?

**Comment by qqiu-cyt** [comment]
Yes, I'm seeing the issue on SDL3. I'm using SDL_SetWindowFullscreen() to enable fullscreen mode. I'll attempt to replicate the problem using testwm and provide an update later.

**Comment by slouken** [comment]
Oh, that actually looks correct to me. testwm sets the viewport to the area of the window not covered by the notch, but the mouse coordinates continue throughout the whole window.

**Comment by qqiu-cyt** [comment]
Could you clarify if SDL_GetWindowPosition() returning (0, 0) is expected behavior for a full-screen window on a screen with a notch? In my tests, it returns (0, 0) instead of (0, 40), leading to an offset in the mouse position relative to the window.

**Comment by slouken** [comment]
I see what you're describing. I'm doing some research now to find out how to extend the window into the notch area in fullscreen mode.

**Comment by qqiu-cyt** [comment]
Just to clarify, I'm not aiming to draw into the notch area in fullscreen mode. What I want to address is the discrepancy in the global mouse position relative to the viewport when the window's position doesn't align with the viewport.

**Comment by slouken** [comment]
Yes, if your window isn't at position 0,0, then the global mouse position and the window relative mouse position will differ. Right now on macOS in non-exclusive fullscreen mode, the application window is bumped down below the notch, so what you're seeing is expected.

**Comment by qqiu-cyt** [comment]
In this scenario, what is the recommended way to determine the mouse position relative to the application window that has been bumped down?

**Comment by slouken** [comment]
SDL_GetMouseState() gives you the mouse position relative to the application window while it has focus.

**Comment by qqiu-cyt** [comment]
That solves the issue I was experiencing. Thanks!

**Comment by slouken** [comment]
You're welcome!

**Comment by ocornut** [comment]
I am little confused by this resolution. Please note that I don't have access to a Mac with a notch, but I suspect there has been a misunderstanding in the question.

The original stated problem was:

> Could you clarify if SDL_GetWindowPosition() returning (0, 0) is expected behavior for a full-screen window on a screen with a notch? In my tests, it returns (0, 0) instead of (0, 40), leading to an offset in the mouse position relative to the window.

And your answer was eventually:

> Yes, if your window isn't at position 0,0, then the global mouse position and the window relative mouse position will differ.

Which AFAIK was understood and expected.

**But basically the current situation seems to be that `SDL_GetGlobalMouseState() - SDL_GetWindowPosition()` is not the same value as the one you'd get from `SDL_EVENT_MOUSE_MOTION` and from `SDL_GetMouseState()`.** Which seems odd, given the two later event/function as advertised as "window relative" position.

> SDL_GetMouseState() gives you the mouse position relative to the application window while it has focus.

Using `SDL_GetMouseState()` works but seems like a workaround to a bug there?

More-over, it would only work for the focused window. 
Does a fullscreen window necessarily has focus? What about when running on multiple screen?

(The reason we have a path using `SDL_GetGlobalMouseState() - SDL_GetWindowPosition()` in imgui backend is to be able to provide "idle" mouse position outside window boundaries **when no button is held, and capture has been made**, which tends to be useful when displaying contents attached to the mouse cursor (e.g. a tooltip, or large image) so it won't be stuck by window edge.)

Links to imgui issues for context
- https://github.com/ocornut/imgui/issues/7786
- https://github.com/ocornut/imgui/pull/7919

**Comment by slouken** [comment]
Okay, I understand the problem now. SDL is assuming that fullscreen windows are at 0,0 which isn’t true in this case. I just got access to a Mac with a notch, so I’ll confirm and work on a fix.

**Comment by slouken** [comment]
It turns out this is an easy fix, we just need to use the window position provided by the OS instead of assuming that the window completely covers the display.

**Comment by ocornut** [comment]
Thank you Sam !

**Comment by slouken** [comment]
You're welcome!
